swap/api.go

```diff
@@ -141,6 +143,8 @@ func (s *Swap) PeerCheques(peer enode.ID) (PeerCheques, error) {
 
 	swapPeer := s.getPeer(peer)
 	if swapPeer != nil {
+		swapPeer.lock.Lock()
+		defer swapPeer.lock.Unlock()
```

Comment:
i prefer that you manually `Unlock` after `L150`. the defer statement will be executed only when the method returns. So if in the future more code will be added below that somehow implicitly uses the same lock a deadlock would occur.

Reply:
makes sense, changed.